packages/react-router/tests/link.test.tsx

```diff
@@ -1,44 +1,805 @@
+    }
+
+    const postRoute = createRoute({
+      getParentRoute: () => postsRoute,
+      path: '$postId',
+      component: PostComponent,
+    })
+
+    const router = createRouter({
+      routeTree: rootRoute.addChildren([
+        indexRoute,
+        postsRoute.addChildren([postRoute]),
+      ]),
+    })
+
+    render(<RouterProvider router={router} />)
+
+    const postLink = await screen.findByRole('link', { name: 'To first post' })
+
+    expect(postLink).toHaveAttribute('href', '/posts/id1')
+
+    fireEvent.click(postLink)
+
+    expect(await screen.findByText('Params: id1')).toBeInTheDocument()
+  })
+
+  test('when navigating from /posts to ./$postId', async () => {
+    const rootRoute = createRootRoute()
+    const indexRoute = createRoute({
+      getParentRoute: () => rootRoute,
+      path: '/',
+      component: () => {
+        return (
+          <React.Fragment>
+            <h1>Index</h1>
+            <Link to="/posts">Posts</Link>
+            <Link to="/posts/$postId" params={{ postId: 'id1' }}>
+              To first post
+            </Link>
+          </React.Fragment>
+        )
+      },
+    })
+
+    const PostsComponent = () => {
+      return (
+        <React.Fragment>
+          <h1>Posts</h1>
+          <Outlet />
+        </React.Fragment>
+      )
+    }
+
+    const postsRoute = createRoute({
+      getParentRoute: () => rootRoute,
+      path: 'posts',
+      component: PostsComponent,
+    })
+
+    const PostsIndexComponent = () => {
+      return (
+        <React.Fragment>
+          <h1>Posts Index</h1>
+          <Link from="/posts/" to="./$postId" params={{ postId: 'id1' }}>
+            To the first post
+          </Link>
+        </React.Fragment>
+      )
+    }
+
+    const postsIndexRoute = createRoute({
+      getParentRoute: () => postsRoute,
+      path: '/',
+      component: PostsIndexComponent,
+    })
+
+    const PostComponent = () => {
+      const params = useParams({ strict: false })
+      return (
+        <React.Fragment>
+          <span>Params: {params.postId}</span>
+          <Link to="/">Index</Link>
+        </React.Fragment>
+      )
+    }
+
+    const postRoute = createRoute({
+      getParentRoute: () => postsRoute,
+      path: '$postId',
+      component: PostComponent,
+    })
+
+    const router = createRouter({
+      routeTree: rootRoute.addChildren([
+        indexRoute,
+        postsRoute.addChildren([postsIndexRoute, postRoute]),
+      ]),
+    })
+
+    render(<RouterProvider router={router} />)
+
+    const postsLink = await screen.findByRole('link', { name: 'Posts' })
+
+    expect(postsLink).toHaveAttribute('href', '/posts')
+
+    fireEvent.click(postsLink)
+
+    expect(await screen.findByText('Posts Index')).toBeInTheDocument()
+
+    const postLink = await screen.findByRole('link', {
+      name: 'To the first post',
+    })
+
+    expect(postLink).toHaveAttribute('href', '/posts/id1')
+    expect(postLink).toHaveAttribute('from', '/posts/')
```

Comment:
Is this checking the `<Link>` component as React node? or the actual HTML that's printed to the document?

If it is the `<Link>` component, then cool 👍🏼.

But if it is the HTML anchor element being printed to the document, then maybe that's actually a bug/unintended behaviour? Since `from` is not a valid HTML Anchor attribute, then it's probably not something we should then be *expecting* to available in the DOM.

Reply:
Yeah, I guess this is a bug!. It's the `a` tag